_episodes/09-cmorization.md

```diff
@@ -0,0 +1,770 @@
+tables are a copy of the [PCMDI](https://github.com/PCMDI) guidelines.
+
+> ## Find the variable "gpp" in a CMOR table
+>
+> Check the available CMOR tables to find the variable "gpp" with the following characteristics:
+> - standard_name: ``gross_primary_productivity_of_biomass_expressed_as_carbon``
+> - frequency: ``mon``
+> - modeling_realm: ``land``
+>
+> > ## Answers
+> >
+> > The variable "gpp" belongs to the land variables. The temporal resolution that we are looking
+> > for is "monthly". This information points to the "Lmon" CMIP table. And indeed, the variable
+> > "gpp" can be found in the file
+> > [here](https://github.com/ESMValGroup/ESMValCore/blob/master/esmvalcore/cmor/tables/cmip6/Tables/CMIP6_Lmon.json).
+> >
+> {: .solution}
+{: .challenge}
+
+If the variable you are interested in is not available in the standard CMOR
+tables, you could write a custom CMOR table entry for the variable. This,
+however, is beyond the scope of this tutorial.
+
+> ## Fill the configuration file
+>
+> Uncomment the following entries in your configuration file and fill them with appropriate values:
+>
+> - dataset_id
+> - version
+> - tier
+> - modeling_realm
+> - short_name (the ??? immediately under `variables`)
+> - mip
+>
+> > ## Answers
+> >
+> > The configuration file now look something like this:
+> >
+> > ```yaml
+> > ---
+> > filename: 'GPP.ANN.CRUNCEPv6.monthly.*.nc'
+> >
+> > attributes:
+> >   project_id: OBS
+> >   dataset_id: FLUXCOM
+> >   version: 'ANN-v1'
+> >   tier: 3
+> >   modeling_realm: reanaly
+> >   source: ''
+> >   reference: ''
+> >   comment: ''
+> >
+> > variables:
+> >   gpp:
+> >     mip: Lmon
+> > ```
+> >
+> {: .solution}
+{: .challenge}
+
+Now that we have set this information correctly in the config file, we can call
+the save function. Add the following python code to your CMORizer script:
+
+```python
+# 3. store the data with the correct filename
+attributes = cfg['attributes']
+variables = cfg['variables']
+
+for short_name, variable_info in variables.items():
+    all_attributes = {**attributes, **variable_info}  # add the mip to the other attributes
+    utils.save_variable(cube=cube, var=short_name, outdir=out_dir, attrs=all_attributes)
+```
+
+Since we only have one variable (gpp), the loop is not strictly necessary.
+However, this makes it possible to add more variables later on.
+
+> ## Was the CMORization successful so far?
+>
+> If you run the CMORizer again, you should see that it creates an output file
+> named ``OBS_FLUXCOM_reanaly_ANN-v1_Lmon_gpp_xxxx01-xxxx12.nc``. The "xxxx" and
+> "yyyy" represent the start and end year of the data.
+>
+{: .callout}
+
+Great! So we have produced a NetCDF file with the CMORizer that follows the
+naming convention for ESMValTool datasets. Let's have a look at the NetCDF file
+as it was written with the very basic CMORizer from above.
+
+```bash
+ncdump -h OBS_FLUXCOM_reanaly_ANN-v1_Lmon_gpp_198001-198012.nc
```

Comment:
And this should be for the year 2000 and not 1980. I had used 1980 as an example before I knew that you, Peter, had instructed people to download the year 2000.

Reply:
Check 👍